inspirehep/records/api/base.py

```diff
@@ -166,3 +179,228 @@ def hard_delete(self):
                 RecordIdentifier.query.filter_by(recid=pid.pid_value).delete()
                 db.session.delete(pid)
             db.session.delete(self.model)
+
+    def get_bucket(self, location=None, storage_class=None, record_id=None):
+        """Allows to retreive bucket for any record(default: self)
+        Args:
+            location (str): Bucket location
+                (default: 'RECORDS_DEFAULT_FILE_LOCATION_NAME') from config
+            storage_class (str): Bucket storage class
+                (default: 'RECORDS_DEFAULT_STORAGE_CLASS') from config
+            record_id (int): record to which bucket is asigned
+                (default: self)
+
+        Returns: Bucket found in db for selected location, storage_class and record_id
+            If there is no bucket for selected parameters, returns None
+
+        """
+        if not storage_class:
+            storage_class = current_app.config["RECORDS_DEFAULT_STORAGE_CLASS"]
+        if not location:
+            location = current_app.config["RECORDS_DEFAULT_FILE_LOCATION_NAME"]
+        if not record_id:
+            record_id = self.id
+        try:
+            location_obj = Location.get_by_name(location)
+        except NoResultFound:
+            raise NoResultFound(
+                f"Cannot find location '{location}'. "
+                f"Please check if system is configured properly!"
+            )
+
+        bucket = (
+            RecordsBuckets.query.join(Bucket)
+            .filter(
+                RecordsBuckets.record_id == record_id,
+                Bucket.default_storage_class == storage_class,
+                Bucket.default_location == location_obj.id,
+            )
+            .one_or_none()
```

Comment:
isn't it enough:

```python
 records_buckets = RecordsBuckets.query.filter_by(
            record_id=self.id).first()
```

Reply:
It's safer this way. With old code I was able to create two buckets for record, and it was making problems. It's safer to be sure that there is only one bucket and we are not saving files to second bucket which might be invisible for the system in some use-cases.